aten/src/ATen/native/mps/operations/Pooling.mm

```diff
@@ -359,6 +359,10 @@ Tensor mps_max_pool2d(const Tensor& input,
                       IntArrayRef padding,
                       IntArrayRef dilation,
                       bool ceil_mode) {
+  TORCH_WARN_ONCE("max_pool2d with `return_indices=False` on MPS is not infinitely differentiable.",
```

Comment:
I think this will be a bit too verbose.
A good way to reduce noise would be to move this to `max_pool2d_backward` and condition it on `at::GradMode::is_enabled()`. This way the warning will only trigger durning the backward pass if `create_graph=True`!
Another option is to register a custom derivative for `max_pool2d_backward` implemented in FunctionsManual.cpp that just raises a nice error.

Reply:
Chose the second option as the first option still raises an unimplemented error when trying to calculate higher order derivatives, which is annoying lol.